File: plugins/reporting/skills/devexpress-reporting-winforms/examples/quickstart.vb
' DevExpress WinForms Reporting — Quickstart (VB.NET)
' Demonstrates: Print preview, designer launch, embedded viewer

Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
Imports DevExpress.LookAndFeel

' 1. Ribbon Print Preview (modal)
Public Class PrintPreviewExample
    Public Shared Sub ShowPreview()
        Dim report As New XtraReport()
        report.DataSource = SampleData.GetProducts()

        Dim detail As New DetailBand()
        detail.HeightF = 25

        Dim label As New XRLabel()
        label.BoundsF = New System.Drawing.RectangleF(0, 2, 300, 20)
        label.ExpressionBindings.Add(New ExpressionBinding("Text", "[Name]"))
        detail.Controls.Add(label)
        report.Bands.Add(detail)

        Dim tool As New ReportPrintTool(report)
        tool.ShowRibbonPreviewDialog(UserLookAndFeel.Default)
    End Sub
End Class

' 2. Ribbon Designer (modal) — Reporting license
Public Class DesignerExample
    Public Shared Sub ShowDesigner()
        Dim report As New XtraReport()
        ' Optional: load existing layout
        ' report.LoadLayoutFromXml("layout.repx")
        ' report.DataSource = SampleData.GetProducts()

        Dim tool As New ReportDesignTool(report)
        tool.ShowRibbonDesignerDialog(UserLookAndFeel.Default)

        report.SaveLayoutToXml("layout.repx")
    End Sub
End Class

' 3. Embedded DocumentViewer
Public Class ReportViewerForm
    Inherits Form

    Private viewer As DocumentViewer

    Public Sub New()
        Me.Text = "Report Viewer"
        Me.Size = New System.Drawing.Size(1000, 700)

        viewer = New DocumentViewer()
        viewer.Dock = DockStyle.Fill
        Controls.Add(viewer)
    End Sub

    Private Sub ReportViewerForm_Load(sender As Object, e As EventArgs) Handles Me.Load
        Dim report As New XtraReport()
        report.DataSource = SampleData.GetProducts()

        Dim detail As New DetailBand()
        detail.HeightF = 25

        Dim label As New XRLabel()
        label.BoundsF = New System.Drawing.RectangleF(0, 2, 300, 20)
        label.ExpressionBindings.Add(New ExpressionBinding("Text", "[Name]"))
        detail.Controls.Add(label)
        report.Bands.Add(detail)

        viewer.DocumentSource = report
        report.CreateDocument()
    End Sub
End Class

' Sample data
Public Module SampleData
    Public Class Product
        Public Property Name As String
        Public Property Price As Decimal
    End Class

    Public Function GetProducts() As System.Collections.Generic.List(Of Product)
        Dim list As New System.Collections.Generic.List(Of Product)()
        list.Add(New Product() With {.Name = "Widget A", .Price = 9.99D})
        list.Add(New Product() With {.Name = "Widget B", .Price = 14.99D})
        Return list
    End Function
End Module

File: plugins/reporting/skills/devexpress-reporting-winforms/examples/quickstart.cs
// DevExpress WinForms Reporting — Quickstart
// Demonstrates: ReportPrintTool (preview), ReportDesignTool (designer), DocumentViewer (embedded)
// Package (Reporting subscription): DevExpress.Win.Reporting

using System;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
using DevExpress.LookAndFeel;
using DevExpress.XtraPrinting.Caching;

// ------------------------------------------------------------------
// 1. Show Ribbon Print Preview (modal)
// ------------------------------------------------------------------
public class PrintPreviewExample
{
    public static void ShowPreview()
    {
        var report = new XtraReport { DataSource = SampleData.GetProducts() };

        var detail = new DetailBand { HeightF = 25 };
        var label = new XRLabel { BoundsF = new System.Drawing.RectangleF(0, 2, 300, 20) };
        label.ExpressionBindings.Add(new ExpressionBinding("Text", "[Name]"));
        detail.Controls.Add(label);
        report.Bands.Add(detail);

        var tool = new ReportPrintTool(report);
        tool.ShowRibbonPreviewDialog(UserLookAndFeel.Default);
    }
}

// ------------------------------------------------------------------
// 2. Launch Ribbon Designer (modal) — Reporting license
// ------------------------------------------------------------------
public class DesignerExample
{
    public static void ShowDesigner()
    {
        var report = new XtraReport();
        report.LoadLayoutFromXml("layout.repx"); // optional: start from saved layout
        report.DataSource = SampleData.GetProducts(); // re-assign after load

        var tool = new ReportDesignTool(report);
        tool.ShowRibbonDesignerDialog(UserLookAndFeel.Default);

        // Save after designer closes
        report.SaveLayoutToXml("layout.repx");
    }
}

// ------------------------------------------------------------------
// 3. Embedded DocumentViewer in a WinForms Form
// ------------------------------------------------------------------
public class ReportViewerForm : Form
{
    private DocumentViewer viewer;

    public ReportViewerForm()
    {
        Text = "Report Viewer";
        Size = new System.Drawing.Size(1000, 700);

        viewer = new DocumentViewer { Dock = DockStyle.Fill };
        Controls.Add(viewer);

        Load += ReportViewerForm_Load;
    }

    private void ReportViewerForm_Load(object sender, EventArgs e)
    {
        var report = new XtraReport { DataSource = SampleData.GetProducts() };

        var detail = new DetailBand { HeightF = 25 };
        var label = new XRLabel { BoundsF = new System.Drawing.RectangleF(0, 2, 300, 20) };
        label.ExpressionBindings.Add(new ExpressionBinding("Text", "[Name]"));
        detail.Controls.Add(label);
        report.Bands.Add(detail);

        viewer.DocumentSource = report;
        report.CreateDocument();
    }
}

// ------------------------------------------------------------------
// 4. Embedded Designer Component (WinForms license)
// ------------------------------------------------------------------
public class EmbeddedDesignerForm : XRDesignRibbonForm
{
    public EmbeddedDesignerForm(XtraReport report)
    {
        OpenReport(report);

        // Optional: configure before showing
        DesignMdiController.DefaultReportSettings.ShowGrid = true;
    }
}

// ------------------------------------------------------------------
// Sample data helper
// ------------------------------------------------------------------
public static class SampleData
{
    public static System.Collections.Generic.List<Product> GetProducts() =>
        new System.Collections.Generic.List<Product>
        {
            new Product { Name = "Widget A", Price = 9.99m },
            new Product { Name = "Widget B", Price = 14.99m },
        };

    public class Product
    {
        public string Name { get; set; }
        public decimal Price { get; set; }
    }
}
